CMakeLists.txt

```diff
@@ -48,7 +48,17 @@ if(HNSWLIB_EXAMPLES)
         endif()
       endif()
     elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-      SET( CMAKE_CXX_FLAGS  "-Ofast -lrt -std=c++11 -DHAVE_CXX0X -march=native -fpic -w -fopenmp -ftree-vectorize -ftree-vectorizer-verbose=0" )
+      SET( CMAKE_CXX_FLAGS  "-Ofast -std=c++11 -DHAVE_CXX0X -fpic -w -fopenmp -ftree-vectorize -ftree-vectorizer-verbose=0" )
+      if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc|powerpc")
+        SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mtune=native" )
+        message("set -mtune=native flag")
+      else()
+        SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native" )
+        message("set -march=native flag")
+      endif()
+      if(NOT APPLE)
```

Comment:
I think we check MacOS several lines above
`if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")`

Do we need to check it here again?

Reply:
What is the related of these two?

Clang can be used on non-Apple systems, and GCC can be used on Apple.

Reply:
Thank you for explanation

Reply:
Clang is used as the default compiler in Xcode since macOS 10.7, but GCC can be used with no issues on any macOS from 10.4 to 14.4.1.
Clang is also a default compiler in the recent versions of FreeBSD.
So by no means it should be seen as “Clang = Apple ”.